### ISSUE TITLE
Wrong path to libxprs.dylib picked up

### ISSUE THREAD

**Issue by dpo** [JuliaOpt/Xpress.jl#8]
For some reason, in Julia 0.6, `Libdl.dlopen` agrees to open a library that doesn't exist causing `Pkg.build` to record the wrong path to `libxprs`:
```julia
julia> Libdl.dlopen("/Users/dpo/local/solvers/xpressmp/bin/lib/libxprs.dylib")
Ptr{Void} @0x00007f9d98004a80
```
but
```bash
dpo:~ $ ls /Users/dpo/local/solvers/xpressmp/bin/lib/libxprs.dylib
ls: /Users/dpo/local/solvers/xpressmp/bin/lib/libxprs.dylib: No such file or directory

dpo:~ $ echo $XPRESS
/Users/dpo/local/solvers/xpressmp/bin

dpo:~ $ echo $XPRESSDIR
/Users/dpo/local/solvers/xpressmp
```
It may be safer to check `XPRESSDIR` instead of `XPRESS` and go look in `$XPRESSDIR/lib`. There's no `lib` subdirectory in `bin`.

**Comment by dpo** [comment]
This works on OSX
```diff
diff --git a/deps/build.jl b/deps/build.jl
index a503f18..f6a2319 100755
--- a/deps/build.jl
+++ b/deps/build.jl
@@ -26,9 +26,8 @@ for a in aliases
             push!(paths_to_try, joinpath(ENV["XPRESSDIR"],"bin",string(a,".",Libdl.dlext)))
         end
     elseif is_unix()
-        if haskey(ENV, "XPRESS")
-            push!(paths_to_try, joinpath(ENV["XPRESS"],"lib",string("lib",a,".",Libdl.dlext)))
-            push!(paths_to_try, joinpath(ENV["XPRESS"],"..","lib",string("lib",a,".",Libdl.dlext)))
+        if haskey(ENV, "XPRESSDIR")
+            push!(paths_to_try, joinpath(ENV["XPRESSDIR"],"lib",string("lib",a,".",Libdl.dlext)))
         end
     end
 end
```
but the next error is
```julia
julia> using Xpress
INFO: Recompiling stale cache file /Users/dpo/.julia/lib/v0.6/Xpress.ji for module Xpress.
INFO: Xpress: Failed to find working license.
ERROR: InitError: Xpress licensing error 1: The license file (/Users/dpo/local/solvers/xpressmp/lib/xpauth.xpr) could not be read. No such file or directory. Please ensure you have installed it according to the instructions in the FICO Xpress READMEs.
```
The Xpress installer puts the license in `$XPRESS`, i.e., `$XPRESSDIR/bin`.

**Comment by dpo** [comment]
Commenting out the license check worked for me. I use a static academic license. The process may be different for other kinds of licenses but the current implementation doesn't work for my kind of license. Thanks for developing this package!

**Comment by joaquimg** [comment]
Good that you are using it. I haven´t tested in OSX for a while, it was working for windows and linux. Unfortunately, I don´t expect to have access to a mac in the next few weeks. @dpo would like to open a PR so that we can converge in a solution that work on linux, osx and widows?

**Comment by joaquimg** [comment]
You can add `is_apple()` in the places that do not work for you

**Comment by dpo** [comment]
Thanks. Will do. Are the directory structure and library path really different and linux and OSX?

**Comment by joaquimg** [comment]
Maybe they should not be different, just do the fix for unix in the PR and I will test on linux then

**Comment by joaquimg** [comment]
Closed by #9
